### DISCUSSION TITLE
Non-binary relationships in OCA

### DISCUSSION THREAD

**Discussion by blelump** [the-human-colossus-foundation/oca-spec#67] [General]
The #63 introduces nonbinary matching into OCA, specifically https://github.com/the-human-colossus-foundation/oca-spec/pull/63#discussion_r1731285606 . It adds SKOS predicates matching that introduces novel terms like close or broad matches, challenging mechanical reasoning. 

What do the proposal authors suggest for potential consumers of this approach to follow? Specifically, what is the suggested meaning of a term that closely matches another term in another ontology? Or is there any, and instead, is it more specific to the use case?

What if mechanical reasoning is a viable option, and i.e., instead of using words with specific meaning, we use proximity or distance that is purely mechanical? 

Can we configure the way we choose the relationship representation?

**Comment by blelump** [comment]
i.e. we could extract predicate metadata, that currently is part of `Predicate_id": "skos:broadMatch"` into `predicate_metadata: { "vocabulary": "skos" }`

**Comment by carlyh-micb** [comment]
An advantage of SKOS using more than just exact:Match: "Lightly specified by design, SKOS is intended to prevent data publishers from introducing false precision into their data and to prevent [inference engines](https://www.sciencedirect.com/topics/computer-science/inference-engines) from drawing unwarranted conclusions." [ref](https://www.sciencedirect.com/science/article/pii/S1570826813000176).

I currently know of no standard where the distance between terms has been formalized and calculated, but that could be done when the standard is finished and using the correct predicate.

We could extract predicate metadata for both predicate_id and framing_justification?

```
{
  "capture_base": "Etszl9LgLUjllI950rd2lO6rF5-BP_jGzXGBPkFZCZFA",
  "type": "spec/overlays/attribute_framing/1.0",
  "framing_metadata": {
    "frame_id": "SNOMEDCT",
    "frame_label": "Systematized Nomenclature of Medicine Clinical Terms",
    "frame_location": "https://bioportal.bioontology.org/ontologies/SNOMEDCT",
    "frame_version": "2023AA"
  },
  "attribute_framing": {
    "Albumin_concentration": {
      "http://purl.bioontology.org/ontology/SNOMEDCT/365801005": {
        "predicate_metadata": {
          "vocabulary": "skos",
          "match_type": "exactMatch"
        },
        "framing_justification": {
          "vocabulary": "semapv",
          "justification": "ManualMappingCuration"
        }
      },
      "http://purl.bioontology.org/ontology/SNOMEDCT/365799007": {
        "predicate_metadata": {
          "vocabulary": "skos",
          "match_type": "broadMatch"
        },
        "framing_justification": {
          "vocabulary": "semapv",
          "justification": "ManualMappingCuration"
        }
      }
    },
    "Glucose_concentration": {
      "http://purl.bioontology.org/ontology/SNOMEDCT/365811003": {
        "predicate_metadata": {
          "vocabulary": "skos",
          "match_type": "broadMatch"
        },
        "framing_justification": {
          "vocabulary": "semapv",
          "justification": "ManualMappingCuration"
        }
      }
    }
  }
}
```

**Reply by blelump** [reply]
Non-binary matches are certainly possible. SKOS is just one of the options, I presume, therefore let's be open for other options.

**Comment by carlyh-micb** [comment]
Typically I see "skos:broadMatch" used, and owl:equivalentClass and owl:equivalentProperty would be other examples using OWL instead of SKOS.
This vocabulary:term structure is something that is very common and would be more recognized by the community than being explicit about the vocabulary as in the latest example given.

In the RFC for framing we addressed this:
"Predicate_id Recommended to use skos terms for the mapping vocabulary (although other mapping schemas would be supported such as owl:)" https://github.com/the-human-colossus-foundation/oca-spec/blob/master/RFCs/0004-framingOverlay.md
And similar for the justification vocabulary.

So there is the opportunity to use other vocabularies for expressing relationships. skos: or owl: or whatever else.

**Reply by carlyh-micb** [reply]
Also, I note that unit framing only talks about SKOS:ExactMatch. Should we update the standard to say or equivalent term in other vocabularies? Similar to attribute framing and entry code framing.
